File: solutions/c/square-root/1/square_root.c
#include "square_root.h"
int square_root(int n) {
    int left = 0;
    int right = n;
    while (left < right) {
        int middle = (left + right + 1) / 2;
        if (n / middle >= middle) {
            left = middle;
        } else {
            right = middle - 1;
        }
    }
    return left;
}

File: solutions/csharp/square-root/1/SquareRoot.cs
public static class SquareRoot
{
    public static int Root(int number)
    {
        var left = 0;
        var right = number;
        while (left < right)
        {
            var middle = (left + right + 1) / 2;
            if (number / middle >= middle)
            {
                left = middle;
            }
            else
            {
                right = middle - 1;
            }
        }
        return left;
    }
}
